### PR TITLE
Multi Dot Fingertip Path

### PR DESCRIPTION
**Pull request by jzhang866**
In mouse_fingertip recognition, use the left-hand gesture 'five' to pause the fingertip trace and break the fingertip tracks into multi-dot paths.

Screen Shot 2023-04-03 at 7 05 13 PM
Screen Shot 2023-04-03 at 7 05 06 PM

### CODE REVIEW

File: fingertip_keyboard_input/mouse_and_keyboard_control.py [hunk 1]
```diff
@@ -397,13 +399,23 @@ def get_filtered_center(center, t):
...
-            # if right hand is pose 'one', append fingertip path
+            # if right hand is pose 'one', and left gesture is not 'five', append the fingertip
+            # if right hand is pose 'one', and left gesture is 'five', pause the fingertip trace
             if right_gesture == 'one':
-                x_pixel = int(right_landmarks.landmark[mp_hands.HandLandmark.INDEX_FINGER_TIP].x\
-                                * image_width)
-                y_pixel = int(right_landmarks.landmark[mp_hands.HandLandmark.INDEX_FINGER_TIP].y\
-                                * image_height)
-                fingertip_path_right.append((x_pixel, y_pixel))
+                if left_gesture == 'five_left' and prev_left_gesture != 'five_left':
```
Comment (gavintchan) [inline fingertip_keyboard_input/mouse_and_keyboard_control.py]: I was testing thing out today, I still think I was accidentally doing other commands with the left hand. For example, I would do:
- left hand = fist (to draw)
- right hand = fist (to submit)

but then accidentally switch to the mouse mode.

Comment (gavintchan) [reply fingertip_keyboard_input/mouse_and_keyboard_control.py]: Changing 405-418 to:
```
if right_gesture != 'one':
                new_fingertip_segment = []
                new_fingertip_segment_appended = False
            if right_gesture == 'one':
                x_pixel = int(right_landmarks.landmark[mp_hands.HandLandmark.INDEX_FINGER_TIP].x\
                                * image_width)

                y_pixel = int(right_landmarks.landmark[mp_hands.HandLandmark.INDEX_FINGER_TIP].y\
                                * image_height)
                new_fingertip_segment.append((x_pixel, y_pixel))
                if new_fingertip_segment_appended and len(fingertip_path_right) > 0:
                    fingertip_path_right[-1] = new_fingertip_segment
                else:
                    fingertip_path_right.append(new_fingertip_segment)
                    new_fingertip_segment_appended = True
```
seems to give intended functionality? I might be missing something though. Let me know what you think / if this was different than what you tried.

https://user-images.githubusercontent.com/60789364/233206475-ddf5b74f-da52-4bee-865e-921eb9764c12.mp4

**Review by gavintchan — COMMENTED** [review]
Multidraw is working well, but proposing some changes for UI side.

**Comment by jzhang866** [comment]
Updated according to Gavin's suggestion, it works well with the pause/append 
Screen Shot 2023-04-26 at 1 22 06 PM

**Comment by gavintchan** [comment]
LGTM, merged.

**Pull request merged**